src/lib/proof-system/zkprogram.ts

```diff
@@ -741,18 +790,41 @@ function ZkProgram<
       } finally {
         snarkContext.leave(id);
       }
+
+      let auxiliaryType = methodIntfs[i].auxiliaryType;
+      let auxiliaryOutputExists =
+        auxiliaryType && auxiliaryType.sizeInFields() !== 0;
+
+      let auxiliaryOutput;
+      if (auxiliaryOutputExists) {
+        let cachedAuxiliaryValue = programState.getAuxiliaryOutput(
+          methodIntfs[i].methodName
+        );
+        auxiliaryOutput = auxiliaryType?.fromValue(cachedAuxiliaryValue);
```

Comment:
why do we do `fromValue()` here? that seems to be a no-op to me because what is returned will be the provable type. `toValue()` would make sense to me though!

Reply:
I changed this